Quantum spanning sub queries now passing

gordonguthrie opened the pull request.

Commit:
The query system now handles sub-queries correctly

Commit:
merge completed

hazen commented on src/riak_kv_qry_compiler.erl:
```diff
@@ -461,7 +470,29 @@ simplest_test() ->
 	    ],
     Expected = [Q#riak_sql_v1{is_executable = true,
                               type          = timeseries,
+<<<<<<< HEAD
```
Fix these merge features, @gordonguthrie, and we can talk.

Commit:
unborking of the merge completed

javajolt commented on src/riak_kv_qry_worker.erl:
```diff
@@ -64,13 +64,14 @@
+          qry      = none      :: none | #riak_sql_v1{},
+          qid      = undefined :: undefined | {node(), non_neg_integer()},
+          n_chunks = 0         :: non_neg_integer(),
+	  sub_qrys = []        :: [integer()],
```
off-by-one (white space)

Commit:
account for multiple 'done' msgs when fetching from quanta-spanning vnodes

There will be multiple (per n_val, redundant) chunk-containing messages
pertaining to a chunk and teerminated by a 'done' message; in
a quanta-spanning fetch, there will be multiple vnodes sending such series
of messages.  Special care needs to be taken to not re-open accumulation
of chunks on arrival of a redundant chunk-containing message when
chunks have already been received from all vnodes.

This has led to indexed, intermediate results being appended to results
already stripped of index, which eventually broke get_column_info
downstream.

Andalso:
* n_chunks field is meaningless in this scenario, thus removed;
* fix capture pattern in all handle_info heads;
* whitespace cleanup.

hazen commented:
Tests now pass

hazen commented:
:+1: 0b0c67d

andytill commented on src/riak_kv_qry_worker.erl:
```diff
@@ -231,13 +248,12 @@ handle_side_effects([{run_sub_query, {qry, Q}, {qid, QId}} | T]) ->
     {ok, _PID} = riak_kv_index_fsm_sup:start_index_fsm(node(), [{raw, QId, Me}, [Bucket, none, Q, Timeout, all, undefined, CoverageFn]]),
     handle_side_effects(T);
 handle_side_effects([H | T]) ->
```
This clause shouldn't exist, it would be a bug if it fell into this clause.

gordonguthrie replied:
I agree

Commit:
response to review comments - remove dead clause

gordonguthrie commented:
+1 999f67e
